added_values.go

```diff
@@ -126,7 +126,7 @@ func (i valueNumber) ToBoolean() bool {
 }
 
 func (i valueNumber) ToObject(r *Runtime) *Object {
-	return r.newPrimitiveObject(i, r.global.NumberPrototype, classNumber)
+	return r.newPrimitiveObject(i, r.getNumberPrototype(), classNumber)
```

Comment:
Pretty much the idea behind object templates is instead of initializing a global value (`r.global.NumberPrototype` in this case) we call a function that returns a singleton instance of a templated object.

[pulling over getNumberPrototype and callees so you don't have to scroll]
```
var numberProtoTemplate *objectTemplate
var numberProtoTemplateOnce sync.Once

func getNumberProtoTemplate() *objectTemplate {
	numberProtoTemplateOnce.Do(func() {
		numberProtoTemplate = createNumberProtoTemplate()
	})
	return numberProtoTemplate
}

func (r *Runtime) getNumberPrototype() *Object {
	ret := r.global.NumberPrototype
	if ret == nil {
		ret = &Object{runtime: r}
		r.global.NumberPrototype = ret
		o := r.newTemplatedObject(getNumberProtoTemplate(), ret)
		o.class = classNumber
	}
	return ret
}
```

Reply:
getter functions that aren't prototypes are a little more straightforward (getArrayValues)
```
func (r *Runtime) getArrayValues() *Object {
	ret := r.global.arrayValues
	if ret == nil {
		ret = r.newNativeFunc(r.arrayproto_values, "values", 0)
		r.global.arrayValues = ret
	}
	return ret
}
```